packages/dds/tree/src/events/events.ts

```diff
@@ -153,22 +154,22 @@ export class EventEmitter<E extends Events<E>> implements IEventEmitter<E> {
      * @returns a function which will deregister the listener when run
      */
     public on<K extends keyof Events<E>>(eventName: K, listener: E[K]): () => void {
-        const listeners = this.listeners[eventName];
+        const listeners = this.listeners.get(eventName);
         if (listeners !== undefined) {
             listeners.add(listener);
         } else {
-            this.listeners[eventName] = new Set([listener]);
+            this.listeners.set(eventName, new Set([listener]));
         }
 
         return this.off.bind(this, eventName, listener);
```

Comment:
personally, I find the alternative style of:
```suggestion
        return () =>this.off(eventName, listener);
```
cleaner. it uses less strange JS features, and is shorter, and is also a pattern that works in more cases than bind does.

Reply:
I agree.